### ISSUE TITLE
 Expected singleton: mail.alias on _alias_is_ascii

### ISSUE THREAD

**Issue by m3asmi** [odoo/odoo#65127]
Impacted versions: 12.0 13.0 14.0

api.constrains must loop over records!

` if self.alias_name and not dot_atom_text.match(self.alias_name):`
we are trying to reach `alias_name` for all the records!

fix:
```

    @api.constrains('alias_name')
    def _alias_is_ascii(self):
        """ The local-part ("display-name" <local-part@domain>) of an
            address only contains limited range of ascii characters.
            We DO NOT allow anything else than ASCII dot-atom formed
            local-part. Quoted-string and internationnal characters are
            to be rejected. See rfc5322 sections 3.4.1 and 3.2.3
        """
        for record in self:
        if self.alias_name and not dot_atom_text.match(self.alias_name):
            raise ValidationError(_("You cannot use anything else than unaccented latin characters in the alias address."))
```

**Comment by pedrobaeza** [comment]
Please propose a pull request with the fix.

**Comment by Yenthe666** [comment]
Plus reproducable steps from a functional perspective would be really nice too :)

**Comment by m3asmi** [comment]
@Yenthe666 it's obvious 
for 12.0 we just add @api.one

**Comment by Yenthe666** [comment]
Now that was a helpful comment...

**Comment by sswapnesh** [comment]
Added Fix on #65159

Step to Reproduce

```
>>>self.env['mail.alias'].search([]).write({'alias_name':'s'})
Traceback (most recent call last):
  File "/home/sswapnesh/WorkSpace/14.0/odoo/models.py", line 4986, in ensure_one
    _id, = self._ids
ValueError: too many values to unpack (expected 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/code.py", line 91, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
  File "/home/sswapnesh/WorkSpace/14.0/addons/mail/models/mail_alias.py", line 122, in write
    return super(Alias, self).write(vals)
  File "/home/sswapnesh/WorkSpace/14.0/odoo/models.py", line 3688, in write
    real_recs._validate_fields(vals, inverse_fields)
  File "/home/sswapnesh/WorkSpace/14.0/odoo/models.py", line 1256, in _validate_fields
    check(self)
  File "/home/sswapnesh/WorkSpace/14.0/addons/mail/models/mail_alias.py", line 90, in _alias_is_ascii
    if self.alias_name and not dot_atom_text.match(self.alias_name):
  File "/home/sswapnesh/WorkSpace/14.0/odoo/fields.py", line 930, in __get__
    record.ensure_one()
  File "/home/sswapnesh/WorkSpace/14.0/odoo/models.py", line 4989, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: mail.alias(1, 2, 3, 4, 5)

```
